pkg/compliance/compliance_test.go

```diff
@@ -31,13 +32,32 @@ func TestCompliance(t *testing.T) {
 	subResReqCount := 0
 
 	apicMockCli.CreateSubResourceMock = func(rm v1.ResourceMeta, subs map[string]interface{}) error {
-		ri, err := agent.GetCacheManager().GetAPIServiceInstanceByName(rm.Name)
-		if ri == nil {
-			return err
+		if rm.GetGroupVersionKind().Kind == management.APIServiceInstanceGVK().Kind {
+			ri, err := agent.GetCacheManager().GetAPIServiceInstanceByName(rm.Name)
+			if ri == nil {
+				return err
+			}
+			ri.SubResources = subs
+			agent.GetCacheManager().AddAPIServiceInstance(ri)
+			subResReqCount++
+		} else if rm.GetGroupVersionKind().Kind == management.EnvironmentGVK().Kind {
+			ri := agent.GetCacheManager().GetWatchResourceByName(management.EnvironmentGVK().Group, management.EnvironmentGVK().Kind, rm.Name)
+			if ri == nil {
+				return errors.New("err")
+			}
+			ri.SubResources = subs
+			agent.GetCacheManager().AddWatchResource(ri)
+			subResReqCount++
+		} else if rm.GetGroupVersionKind().Kind == management.APIServiceGVK().Kind {
+			ri := agent.GetCacheManager().GetWatchResourceByName(management.APIServiceGVK().Group, management.APIServiceGVK().Kind, rm.Name)
+			if ri == nil {
+				return errors.New("err")
+			}
+			ri.SubResources = subs
+			agent.GetCacheManager().AddWatchResource(ri)
+			subResReqCount++
 		}
```

Comment:
a lot of this looks like the same code with a diff ri value, any way to simplify this logic?